accounts-db/src/tiered_storage/footer.rs

```diff
@@ -200,13 +232,21 @@ impl TieredStorageFooter {
         let mut footer = Self::default();
         file.seek_from_end(-(footer_size as i64))?;
         file.read_type(&mut footer)?;
+        Self::sanitize(&footer)?;
 
         Ok(footer)
     }
 
     pub fn new_from_mmap(mmap: &Mmap) -> TieredStorageResult<&TieredStorageFooter> {
         let offset = mmap.len().saturating_sub(FOOTER_TAIL_SIZE);
-        let (footer_size, offset) = get_type::<u64>(mmap, offset)?;
+        let (&footer_size, offset) = get_type::<u64>(mmap, offset)?;
+        if footer_size != FOOTER_SIZE as u64 {
```

Comment:
Same here.  The size might be footer-format-version-dependent.  Probably better to have one function handle this.